Add multiarchitecture CSI plugins

jamesorlakin opened the pull request:
This is an implementation to close #79 using the same techniques for [my unofficial images](https://hub.docker.com/u/jamesorlakin).

I've built the latest version of each plugin. I haven't updated the main README yet as I presume the images will need to be built for badges first. I also need to track down relevant tracking issues for each plugin...

The primary usecase for this is a Rook+Ceph-based ARM cluster. Rook's operator has a native `amd64` and `arm64` multiarch image. The Ceph CSI plugin is available for both these architectures as well but using a tag-suffix pattern, which makes running Rook autonomously a bit trickier in a mixed architecture cluster.

So with regards to Ceph, do we:
- Pull the official images from Quay, retag, push, and produce a multiarch manifest?
- *Or*, build the images ourselves and try and include `armhf` in this too?

If going for the latter we'd probably want to build Rook with `armhf` as well.

---

All comments welcome, I'm sure I've slipped up somewhere!

xUnholy commented:
@jamesorlakin nice work looks great! With regards to Ceph, I think whatever you think is the best and probably easiest solution is fine to go with.

> I haven't updated the main README yet as I presume the images will need to be built for badges first.

It's okay to add them now, they will work because this PR would have built the images with a commit SHA tag.

xUnholy commented on build/csi-external-snapshotter/build.make.patch:
```diff
@@ -0,0 +1,11 @@
+--- build.make	2020-05-23 16:43:42.192359292 +0100
```
Was this file suppose to be checked in? OR perhaps it still seems to have a diff in the file output

jamesorlakin replied:
Aah indeed - I forgot to mention it! This project's Makefile happens to override the GOOS variable and seemed to always produce an `amd64` binary. Debugging a running build in `buildx` was tricky so this was the workaround I did locally a while ago. I presume there's something smarter we could try doing instead - Would you know where I could feed in [the BUILD_PLATFORMS variable](https://github.com/kubernetes-csi/external-snapshotter/blob/v2.1.1/release-tools/build.make#L70)? I think it just sets it to empty there.

The (builder) Dockerfile for this one installs `patch` and applies it to the checked out repo, before continuing on like the others.

xUnholy commented:
One last request would be to see if we can add these specific images into this workflow https://github.com/raspbernetes/multi-arch-images/blob/master/.github/workflows/schedule.yml so we can maintain release versions as the upstream issue newer releases.

This is a fantastic effort - will be handy for a lot of k8s homelab users!

jamesorlakin commented:
Ooh that schedule looks fancy! Sure I'll take a look.

xUnholy commented:
> Ooh that schedule looks fancy! Sure I'll take a look.

Yeah it's a way to get the latest release from the upstream, then commit the version which will trigger the workflow to build the new release version. 

You can also use custom event triggers but you need a consistent event from the upstream which just isn't viable unfortunately, so this is a best effort without having to manually update new release versions.

jamesorlakin commented:
I've added the schedule scripts and tinkered with `cut` to trim the leading `v` from the version name. I've also added the rows into the README.

For Ceph I'll leave that to another PR if that sounds okay. Still got to decide between retagging or rebuilding...
